library/ssl_tls13_keys.c

```diff
@@ -353,26 +370,37 @@ int mbedtls_ssl_tls13_evolve_secret(
 
     status = psa_key_derivation_setup( &operation,
                                        PSA_ALG_HKDF_EXTRACT( hash_alg ) );
-    if (status == PSA_SUCCESS)
-        status |= psa_key_derivation_input_bytes( &operation,
-                                                  PSA_KEY_DERIVATION_INPUT_SALT,
-                                                  tmp_secret,
-                                                  hlen );
-    if (status == PSA_SUCCESS)
-        status |= psa_key_derivation_input_bytes( &operation,
-                                                  PSA_KEY_DERIVATION_INPUT_SECRET,
-                                                  tmp_input,
-                                                  ilen );
-    if (status == PSA_SUCCESS)
-        status |= psa_key_derivation_output_bytes( &operation,
-                                                   secret_new,
-                                                   PSA_HASH_LENGTH( hash_alg ) );
-    if (status == PSA_SUCCESS)
-        status |= psa_key_derivation_abort( &operation );
-
-    ret = psa_ssl_status_to_mbedtls ( status );
- cleanup:
 
+    if( status != PSA_SUCCESS )
+         goto cleanup;
+
+    status = psa_key_derivation_input_bytes( &operation,
+                                             PSA_KEY_DERIVATION_INPUT_SALT,
+                                             tmp_secret,
+                                             hlen );
+
+    if( status != PSA_SUCCESS )
+         goto cleanup;
+
+    status = psa_key_derivation_input_bytes( &operation,
+                                             PSA_KEY_DERIVATION_INPUT_SECRET,
+                                             tmp_input,
+                                             ilen );
+
+    if( status != PSA_SUCCESS )
+         goto cleanup;
+
+    status = psa_key_derivation_output_bytes( &operation,
+                                              secret_new,
+                                              PSA_HASH_LENGTH( hash_alg ) );
+
+    if( status != PSA_SUCCESS )
```

Comment:
Same as above - this goto doesn't change anything, but it does no harm either :)